src/main/java/seedu/gamebook/logic/commands/EditCommand.java

```diff
@@ -93,16 +97,25 @@ public CommandResult execute(Model model) throws CommandException {
         GameEntry gameEntryToEdit = lastShownList.get(index.getZeroBased());
         GameEntry editedGameEntry = createEditedGameEntry(gameEntryToEdit, editGameEntryDescriptor);
 
-        // TODO - might remove if we are not checking for identical game entries
-        if (!gameEntryToEdit.isSameGameEntry(editedGameEntry) && model.hasGameEntry(editedGameEntry)) {
-            throw new CommandException(MESSAGE_DUPLICATE_GAME);
+        if (gameEntryToEdit.equals(editedGameEntry)) {
+            throw new CommandException(MESSAGE_FIELDS_ARE_IDENTICAL);
         }
+        String sameEntryAlert = model.hasGameEntry(editedGameEntry)
+                ? MESSAGE_DUPLICATE_GAME_ENTRY
+                : "";
+        String inFutureAlert = editedGameEntry.getDate().isInFuture()
+                ? MESSAGE_GAME_OCCURS_IN_FUTURE
+                : "";
 
         model.setGameEntry(gameEntryToEdit, editedGameEntry);
+        model.updateFilteredGameEntryList(PREDICATE_SHOW_ALL_GAME_ENTRIES);
```

Comment:
Not sure if we should remove this since keeping this would mean that the selected list gets cleared (original list will be shown) whenever an edit command is executed.

Reply:
Agree to remove